metricbeat/module/jolokia/jmx/config.go

```diff
@@ -178,3 +459,59 @@ func buildRequestBodyAndMapping(mappings []JMXMapping) ([]byte, AttributeMapping
 	content, err := json.Marshal(blocks)
 	return content, responseMapping, err
 }
+
+// Fetch perfrorms a POST request to Jolokia server and gets information about MBeans.
+func (pc *JolokiaHTTPPostBuilder) Fetch(m *MetricSet) ([]common.MapStr, error) {
+
+	// Prepare Http POST request object and attribute mappings according to selected Http method
+	httpReqs, mapping, err := pc.BuildRequestsAndMappings(m.mapping)
+	if err != nil {
+		return nil, err
+	}
+
+	// Log request information
+	if logp.IsDebug(metricsetName) {
+		for _, r := range httpReqs {
+			m.log.Debugw("Jolokia request URI and body",
+				"httpMethod", r.HTTPMethod, "URI", r.URI, "body", string(r.Body), "type", "request")
+		}
+	}
+
+	http, err := helper.NewHTTP(m.BaseMetricSet)
+
+	http.SetMethod(httpReqs[0].HTTPMethod)
+	http.SetBody(httpReqs[0].Body)
+
+	resBody, err := http.FetchContent()
+	if err != nil {
+		return nil, err
+	}
+
+	if logp.IsDebug(metricsetName) {
+		m.log.Debugw("Jolokia response body",
+			"host", m.HostData().Host, "uri", http.GetURI(), "body", string(resBody), "type", "response")
+	}
+
+	// Construct a new event mapper
+	eventMapper := NewEventMapper(httpReqs[0].HTTPMethod)
+
+	// Map response to Metricbeat events
+	events, err := eventMapper.EventMapping(resBody, mapping)
```

Comment:
Is there any reason to don't make these EventMapping methods of `JolokiaHTTPPostBuilder` and `JolokiaHTTPGetBuilder` directly?

Reply:
Hi @jsoriano ! Indeed, there is no reason. I moved `EventMapping` methods to `JolokiaHTTPGetFetcher` and `JolokiaHTTPPostFetcher`. :+1: